gunrock/util/kernel_segmode.hxx

```diff
@@ -0,0 +1,1259 @@
+
+#pragma once
+
+// imports from mgpu
+#include <moderngpu/cta_scan.hxx>
+#include <moderngpu/context.hxx>
+#include <moderngpu/loadstore.hxx>
+#include <moderngpu/cta_search.hxx>
+#include <moderngpu/cta_segsort.hxx>
+#include <moderngpu/cta_load_balance.hxx>
+#include <moderngpu/memory.hxx>
+
+namespace gunrock {
+namespace util {
```

Comment:
All of your functions will be in the util namespace. If you want to have some of your helper functions separate from the rest of the util functions, you should place them in an anonymous namespace. (See https://stackoverflow.com/a/3363730). This can help logically separate the functions you're exposing to the user and the ones you are not.